### ISSUE TITLE
NSRangeException: *** -[__NSArrayM getObjects:range:]: range {0, 5} extends beyond bounds [0 .. 3]

### ISSUE THREAD

**Issue by edenman** [BranchMetrics/ios-branch-deep-linking#622]
Sounds like it's probably a race condition when persisting the request queue to disk?
Branch pod 0.14.12
iPhone 7Plus running iOS 10.3.1
```
Application Specific Information:
Application threw exception NSRangeException: *** -[__NSArrayM getObjects:range:]: range {0, 5} extends beyond bounds [0 .. 3]

Thread 0 name: 
Thread 0 Crashed:
0   CoreFoundation                  0x30beb8fd8         ___exceptionPreprocess
1   libobjc.A.dylib                 0x3094a0538         _objc_exception_throw
2   CoreFoundation                  0x30bd9b12c         -[__NSArrayM getObjects:range:]
3   CoreFoundation                  0x30bd9ac70         -[NSArray initWithArray:range:copyItems:]
4   Branch                          0x1009e08ac         -[BNCServerRequestQueue persistToDisk] (BNCServerRequestQueue.m:209)
5   Foundation                      0x30d4c3760         ___NSFireTimer
6   CoreFoundation                  0x30be67a9c         ___CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__
7   CoreFoundation                  0x30be677a0         ___CFRunLoopDoTimer
8   CoreFoundation                  0x30be67060         ___CFRunLoopDoTimers
9   CoreFoundation                  0x30be64c84         ___CFRunLoopRun
10  CoreFoundation                  0x30bd94d94         _CFRunLoopRunSpecific
11  GraphicsServices                0x30f264074         _GSEventRunModal
12  UIKit                           0x318299130         _UIApplicationMain
13  recharge-consumer-ios           0x2001b7004         main (AppDelegate.swift:44)
14  libdyld.dylib                   0x309db659c         _start
```

**Comment by E-B-Smith** [comment]
That's exactly it: A race condition I've fixed this in the latest version, version 0.15.0.

Version 0.15.0 is in beta release right now if you want to try it out.  It solves this problem along with some other race conditions and other issues discovered while fixing the unit tests.

The Branch iOS Beta program: https://branch.app.link/Beta-Info

**Comment by edenman** [comment]
I'll just wait for it to get to public release.  Thanks!
